src/cares_wrap.cc

```diff
@@ -525,6 +544,573 @@ class QueryWrap : public AsyncWrap {
   }
 };
 
+Local<Array> AddrTTLToArray(Environment* env,
+                            ares_addrttl* addrttls,
+                            int naddrttls) {
+  auto isolate = env->isolate();
+  EscapableHandleScope escapable_handle_scope(isolate);
+
+  Local<Array> ttls = Array::New(isolate, naddrttls);
+  auto context = env->context();
+  for (int i = 0; i < naddrttls; i += 1) {
+    auto value = Integer::New(isolate, addrttls[i].ttl);
+    ttls->Set(context, i, value).FromJust();
+  }
+
+  return escapable_handle_scope.Escape(ttls);
+}
+
+
+Local<Array> AddrTTL6ToArray(Environment* env,
+                             ares_addr6ttl* addrttls,
+                             int naddrttls) {
```

Comment:
Ditto – I think it’s safe to merge these two methods into a template, by the way.